ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -1969,6 +1987,38 @@ public void close() throws IOException {
     dynamicVars.clear();
   }
 
+  private static void cleanUpAllSessionStates() {
+    ExecutorService cleanupExecutor = Executors.newCachedThreadPool();
+    try {
+      CompletableFuture<Void> allCleanupTasks = CompletableFuture.allOf(sessionStates.stream()
+              .map(sessionState -> CompletableFuture.runAsync(() -> {
+                  try {
+                      LOG.info("Closing session state: {}", sessionState.getSessionId());
+                      sessionState.close();
+                  } catch (IOException e) {
+                      LOG.error("Problem closing session state", e);
+                  }
+              }, cleanupExecutor).exceptionally(e -> {
+                  LOG.error("Error closing session state", e);
+                  return null;
+              })).toArray(CompletableFuture[]::new));
+      try {
+        allCleanupTasks.get(60, TimeUnit.SECONDS);
+      } catch (Exception e) {
+        LOG.error("Failed to close all session states", e);
+      }
+    } finally {
+      cleanupExecutor.shutdown();
```

Comment:
why do you need a graceful shutdown if at this point all tasks either completed successfully or timed out?